### PR TITLE
Add warning when mirrord exec docker is detected, hinting the user towards mirrord container.

### PR DESCRIPTION
**Pull request by meowjesty**
- Issue: #2599 

- Adds a new warning when the user does `mirrord exec docker`;
  - Thought about making the message a bit longer, explaining what's going on (similar to the "multi-pod" warning), but it feels like too much text;

### CODE REVIEW

File: mirrord/progress/src/messages.rs [hunk 1]
```diff
@@ -11,3 +11,9 @@ pub const MULTIPOD_WARNING: (&str, &str) = (
         You can get started with mirrord for Teams at this link: \
         https://mirrord.dev/docs/overview/teams/",
 );
+
+pub const EXEC_DOCKER_BINARY: (&str, &str) = (
```
Comment (aviramha) [inline mirrord/progress/src/messages.rs:15]: why the `(&str, &str)` format? I don't see `.0` being used anywhere

**Comment by Razz4780** [comment]
Something weird happened in the `test_agent` job :eyes:

**Comment by meowjesty** [comment]
> Something weird happened in the test_agent job 👀

What? How?

**Comment by Razz4780** [comment]
> > Something weird happened in the test_agent job 👀
> 
> What? How?

Aliens

File: mirrord/progress/src/messages.rs [hunk 2]
```diff
@@ -11,3 +11,6 @@ pub const MULTIPOD_WARNING: (&str, &str) = (
         You can get started with mirrord for Teams at this link: \
         https://mirrord.dev/docs/overview/teams/",
 );
+
+pub const EXEC_DOCKER_BINARY: &str = "`mirrord exec docker` detected! \
+    Please use `mirrord container [options] exec` instead.";
```
Comment (Razz4780) [inline mirrord/progress/src/messages.rs:16]:
```suggestion
pub const EXEC_DOCKER_BINARY: &str = "`mirrord exec <docker|podman|nerdctl> ...` detected! \
    If you try to run a container with mirrord, please use `mirrord container [options] -- <docker|podman|nerdctl> ...` instead.";
```

File: mirrord/cli/src/main.rs [hunk 3]
```diff
@@ -321,6 +321,10 @@ async fn exec(args: &ExecArgs, watch: drain::Watch) -> Result<()> {
         args.binary, args.binary_args
     );
 
+    if args.binary.contains("docker") {
```
Comment (Razz4780) [inline mirrord/cli/src/main.rs:324]:
```suggestion
    if args.binary.contains("docker") || args.binary.contains("podman") || args.binary.contains("nerdctl") {
```

**Review by Razz4780 — CHANGES_REQUESTED** [review]
1. I'd handle all container CLIs we support with `mirrord container ...`
2. Small fix in suggested command line

File: mirrord/cli/src/main.rs (hunk 3 above)
Comment (meowjesty) [reply mirrord/cli/src/main.rs]: Added it as a regex, so we can just add `|whatever`, wdyt? (also is the regex ok? it'll match on things like `foo-docker`)

File: mirrord/cli/src/main.rs [hunk 4]
```diff
@@ -321,6 +322,12 @@ async fn exec(args: &ExecArgs, watch: drain::Watch) -> Result<()> {
         args.binary, args.binary_args
     );
 
+    let container_detection = Regex::new(".*(docker|podman|nerdctl).*")
```
Comment (Razz4780) [inline mirrord/cli/src/main.rs:325]:
```suggestion
    let container_detection = Regex::new("docker|podman|nerdctl")
```
Should be the same (doesn't have to match full string)

**Review by Razz4780 — APPROVED** [review]

**Pull request merged**